### ISSUE TITLE
AWS CodeCommit error MaximumRepositoryNamesExceededException

### ISSUE THREAD

**Issue by lalau** [sourcegraph/sourcegraph#968]
- **Sourcegraph version:** 2.13.2
- **Platform information:** localhost Mac OSX 10.12.6, Docker version 18.09.0-ce-beta1, build 78a6bdb, Google Chrome 70

#### Steps to reproduce:

1. Enter AWS CodeCommit config

#### Expected behavior:

Load AWS CodeCommit repos

#### Actual behavior:

Following error is logged in docker:
```
19:33:57   repo-updater | t=2018-11-06T19:33:57+0000 lvl=eror msg="Error listing AWS CodeCommit repositories" error="MaximumRepositoryNamesExceededException: repositoryNames has too many repository names.  Maximum allowed is 25\n\tstatus code: 400, request id: e7877e0a-e1fa-11e8-a717-a31dc9b4bb1a"
```

Please note that I opened the same issue before:
https://github.com/sourcegraph/sourcegraph/issues/835

The issue was marked closed by bumping down the `maxCount` to 25 in this PR:
https://github.com/sourcegraph/sourcegraph/pull/857/files

But I don't think it's having any effect, when I look further into the code I don't see `maxCount` being used anywhere in the `ListRepositories` function:
https://github.com/sourcegraph/sourcegraph/blob/56bad9d6fb4b40d49dc0e698261afc1efb7b8c35/pkg/extsvc/awscodecommit/repos.go#L150

Also please note that the limit isn't on the aws codecommit  `ListRepositories` api, It's the `BatchGetRepositories` having the limit. You can verify it by aws cli:
ref:
https://docs.aws.amazon.com/cli/latest/reference/codecommit/list-repositories.html
https://docs.aws.amazon.com/cli/latest/reference/codecommit/batch-get-repositories.html

Please use the `maxCount` passed in and apply it on the BatchGetRepositories call (of course with pagination to actually get all repositories).

**Comment by nicksnyder** [comment]
Sorry about that, you are completely right. I just assumed that the implementation already respected that config value.

**Comment by nicksnyder** [comment]
The fix for this was released in 2.13.5. Can you confirm that it works for you?
